accelerator/core/mam_core.c

```diff
@@ -285,27 +329,57 @@ static status_t create_channel_fetch_all_transactions(const iota_client_service_
  * @return return code
  */
 static status_t ta_mam_written_msg_to_bundle(const iota_client_service_t *const service, mam_api_t *const api,
-                                             const size_t channel_depth, mam_encrypt_key_t mam_key,
+                                             const struct channel_info_s *channel_info, mam_encrypt_key_t mam_key,
                                              char const *const payload, bundle_transactions_t **bundle,
                                              tryte_t *const chid, tryte_t *const msg_id,
                                              mam_send_operation_t *mam_operation) {
   status_t ret = SC_OK;
-  if (!service || !api || !chid || !msg_id || channel_depth < 1) {
+  if (!service || !api || !chid || !msg_id || !channel_info || channel_info->ch_mss_depth < 1) {
     ret = SC_MAM_NULL;
     ta_log_error("%s\n", ta_error_to_string(ret));
     return ret;
   }
   trit_t msg_id_trits[MAM_MSG_ID_SIZE];
   hash81_array_p tag_array = hash81_array_new();
+
+  // Get to the assigned beginning channel ID. If the initial setting is different, then tangle-accelerator won't be
+  // able to generate same chid.
+  while (channel_info->chid && memcmp(channel_info->chid, chid, NUM_TRYTES_ADDRESS)) {
+    if (chid[0]) {
+      if (mam_api_channel_create(api, channel_info->ch_mss_depth, chid) != RC_OK) {
+        ret = SC_MAM_FAILED_CREATE_OR_GET_ID;
+        ta_log_error("%s\n", ta_error_to_string(ret));
+        goto done;
+      }
+    } else {
+      bool same_setting = false;
```

Comment:
`same_setting` is a bit confusing. What's that?

Reply:
Differnt setting of MAM (i.e. different MSS depth) would cause different set of channel ID. The function, `same_mam_setting()` is serving for finding whether the current setup of MAM has been used or not.

Reply:
How about renaming it to `unchanged_setting`?

Reply:
OK